src/export/ExportFFmpeg.cpp

```diff
@@ -874,47 +981,34 @@ bool ExportFFmpeg::Finalize()
 
 void ExportFFmpeg::FreeResources()
 {
-   // Close the codecs.
-   mEncAudioCodecCtx.reset();
-
-   // Close the output file if we created it.
-   mUfileCloser.reset();
-
-   // Free any buffers or structures we allocated.
-   mEncFormatCtx.reset();
-
-   mEncAudioFifoOutBuf.reset();
-   mEncAudioFifoOutBufSiz = 0;
 
-   mEncAudioFifo.reset();
-
-   av_log_set_callback(av_log_default_callback);
```

Comment:
You deleted all the resets.  The lifetime of side structures therefore continues until the end of the lifetime of the ExportFFmpeg object.  Did you check that this lifetime isn't longer than when this function is called?  Are you sure the ExportFFmpeg object isn't reused and might expect a clean state when it is?

I tried to determine that from the code, and it's not trivial to prove it.  So maybe leave the harmless release() lines.

Reply:
I don't really see why logger should live shorted than FFmpeg. If it does - we will be losing potentially useful information from the log.